File: src/Timeslot.java
package src;
import java.time.LocalDateTime;

class Slot {
    private LocalDateTime slotStart;
    private LocalDateTime slotEnd;
    private boolean slotAvailability;
    private String slotAssignment;

    public Slot() {
        slotStart = null;
        slotEnd = null;
        slotAvailability = false; // This is showing whether its filled or not, false meaning, it can be assigned a job.
        slotAssignment = null;
    }

    // Getters for Start Time, End Time, Availability, and Assignment
    public LocalDateTime getStart() {
        return slotStart;
    }
    public LocalDateTime getEnd() {
        return slotEnd;
    }
    public boolean getAvailability() {
        return slotAvailability;
    }
    public String getAssignment() {
        return slotAssignment;
    }

    // Setters for Start Time, End Time, Availability, and Assignment
    public void setStart(LocalDateTime start) {
        this.slotStart = start;
    }
    public void setEnd(LocalDateTime end) {
        this.slotEnd = end;
    }
    public void setAvailability(boolean availability) {
        this.slotAvailability = availability;
    }
    public void setAssignment(String assignment) {
        this.slotAssignment = assignment;
    }
}

public class Timeslot {
    public static void main(String[] args) {

    }
}

File: src/Timeslot.py
from datetime import datetime

class Slot:
    def __init__(self):
        self.slotStart = None
        self.slotEnd = None
        self.slotAvailability = False  # This shows whether it's filled or not, False meaning it can be assigned a job.
        self.slotAssignment = None

    # Getters for Start Time, End Time, Availability, and Assignment
    def get_start(self):
        return self.slotStart

    def get_end(self):
        return self.slotEnd

    def get_availability(self):
        return self.slotAvailability

    def get_assignment(self):
        return self.slotAssignment

    # Setters for Start Time, End Time, Availability, and Assignment
    def set_start(self, start):
        self.slotStart = start

    def set_end(self, end):
        self.slotEnd = end

    def set_availability(self, availability):
        self.slotAvailability = availability

    def set_assignment(self, assignment):
        self.slotAssignment = assignment
